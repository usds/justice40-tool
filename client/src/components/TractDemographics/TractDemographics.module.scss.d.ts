declare namespace TractDemographicsNamespace {
    export interface ITractDemographicsScss {
      demographicsContainer: string;
      demographicsTitle: string;
      demographicHeading: string;
      showHideText: string;
      showHideIcon: string;
      demographicItem: string;
    }
  }

declare const TractDemographicsScssModule: TractDemographicsNamespace.ITractDemographicsScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: TractDemographicsNamespace.ITractDemographicsScss;
  };

  export = TractDemographicsScssModule;
