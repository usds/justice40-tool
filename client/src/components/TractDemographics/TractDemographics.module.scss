@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

.demographicsContainer{
    padding: 1.2rem 1rem 0 1.2rem;
    
    .demographicsTitle {
        @include sidePanelLabelStyle;
    }

    .demographicItem {
        display: flex;
        justify-content: space-between;
        padding-bottom: .5rem;
    
        span {
            font-size: .8rem;
            margin-top: 0;
        }
    }

    .demographicItem:first-child {
        @include u-margin-top(.5);
    }
    .demographicHeading {
        @include u-margin-top(1);
        font-size: .875rem;
        
        span{
            @include u-margin-left(1);
        }

        .showHideText {
            //remove styling of button
            border: none;
            background-color: white;
    
            //emulate a link
            cursor:pointer;
            text-decoration:underline;
    
            min-width: 45px;
            padding-left: 0;
        }

        .showHideIcon{
            vertical-align: middle;
            height: 1rem;
            filter: invert(8%) sepia(90%) saturate(7490%) hue-rotate(247deg) brightness(105%) contrast(143%);

            &:hover{
                cursor: pointer;
            }
        }

    }
}
