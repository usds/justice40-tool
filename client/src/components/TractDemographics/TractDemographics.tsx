import React, {useState} from 'react';

// @ts-ignore
import expandIcon from '/node_modules/uswds/dist/img/usa-icons/expand_more.svg';
// @ts-ignore
import collapseIcon from '/node_modules/uswds/dist/img/usa-icons/expand_less.svg';

import * as styles from './TractDemographics.module.scss';
import * as constants from '../../data/constants';
import * as EXPLORE_COPY from '../../data/copy/explore';

export interface ITractDemographicsProps {
  properties: constants.J40Properties
}

interface IDemographicsData {
  race: [React.ReactElement, number][],
  age: [React.ReactElement, number][],
}

interface IJ40AccordionItem {
  id: string,
  title: React.ReactElement,
  children: React.ReactElement
}


/**
 * This function will create the custom Accordion item. This will be used
 * for the race and age demographic UI elements
 *
 * @param {IJ40AccordionItem} props
 * @return {JSX.Element}
 */
const J40AccordionItem = ({id, title, children}:IJ40AccordionItem) => {
  const [isExpanded, setIsExpanded] = useState(false);
  return (
    <>
      <h6 className={styles.demographicHeading}>
        {title}
        <span>
          {'('}
          <a
            className={styles.showHideText ? `usa-link ${styles.showHideText}` : `usa-link`}
            id={`${id}-header`}
            aria-controls={`${id}-panel`}
            aria-expanded={isExpanded}
            onClick={() => setIsExpanded(!isExpanded)}
            tabIndex={0}
          >
            {isExpanded ? 'hide' : 'show'}
          </a>
          { isExpanded ?
          <img
            className={styles.showHideIcon}
            src={collapseIcon}
            alt={'collapse icon'}
            onClick={() => setIsExpanded(!isExpanded)}
          /> :
          <img
            className={styles.showHideIcon}
            src={expandIcon}
            alt={'expand icon'}
            onClick={() => setIsExpanded(!isExpanded)}
          />
          }
          {')'}
        </span>

      </h6>

      <section
        id={`${id}-panel`}
        aria-labelledby={`${id}-header`}
        hidden={!isExpanded}
      >{children}</section>
    </>
  );
};


/**
 * This function will create each line item on the list of demographics
 *
 * @param {[]} demographicData
 * @return {JSX.Element}
 */
const demographicItemGen = (demographicData: []) => {
  return demographicData.map((el, index) => {
    return (
      <div key={index} className={styles.demographicItem}>
        <span>{ el[0] }</span>
        {typeof el[1] === 'number' ?
        <span>{`${el[1]}%`}</span> :
        <span>{`${el[1]}`}</span> }
      </div>
    );
  });
};


/**
 * This function will return the numeric value of each demographic. Taking into
 * account cases when the data is undefined or is null
 *
 * @param {number} stat
 * @return {number}
 */
const displayStat = (stat: number) => {
  if (stat === undefined || stat === null) {
    return '--';
  } else if (stat === 0) {
    return 0;
  }

  return Number(Math.floor(stat * 100));
};


/**
 * This function will create the data structure for the demographics data
 *
 * @param {constants.J40Properties} properties
 * @return {IDemographicsData}
 */
const getDemographicsData = (properties:constants.J40Properties):IDemographicsData => (
  {
    race: [
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_NON_HISPANIC_WHITE,
        displayStat(properties[constants.DEMO_NON_HISPANIC_WHITE]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_BLACK,
        displayStat(properties[constants.DEMO_BLACK]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_AMERICAN_INDIAN,
        displayStat(properties[constants.DEMO_AMERICAN_INDIAN]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_ASIAN,
        displayStat(properties[constants.DEMO_ASIAN]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_HAWAIIAN,
        displayStat(properties[constants.DEMO_HAWAIIAN]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_OTHER_RACE,
        displayStat(properties[constants.DEMO_OTHER_RACE]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_TWO_OR_MORE_RACES,
        displayStat(properties[constants.DEMO_TWO_OR_MORE_RACES]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_HISPANIC,
        displayStat(properties[constants.DEMO_HISPANIC]),
      ],
    ],
    age: [
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_AGE_UNDER_10,
        displayStat(properties[constants.DEMO_AGE_UNDER_10]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_AGE_MID,
        displayStat(properties[constants.DEMO_AGE_MID]),
      ],
      [
        EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.DEMO_AGE_OVER_65,
        displayStat(properties[constants.DEMO_AGE_OVER_64]),
      ],
    ],
  });


/**
 * This is the main component for this file. It accepts the selected feature
 * as a prop and return the demographics component.
 *
 * @param {ITractDemographicsProps} props
 * @return {JSX.Element}
 */
const TractDemographics = ({properties}: ITractDemographicsProps) => {
  const {race, age} = getDemographicsData(properties);

  return (
    <div className={styles.demographicsContainer}>
      <div className={styles.demographicsTitle}>
        {EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.TITLE}
      </div>
      <>
        <J40AccordionItem id={'race'} title={EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.RACE_TITLE}>
          {demographicItemGen(race)}
        </J40AccordionItem>
        <J40AccordionItem id={'age'} title={EXPLORE_COPY.SIDE_PANEL_DEMOGRAPHICS.AGE_TITLE}>
          {demographicItemGen(age)}
        </J40AccordionItem>
      </>
    </div>
  );
};

export default TractDemographics;
