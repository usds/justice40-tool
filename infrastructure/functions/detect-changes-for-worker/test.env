REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1