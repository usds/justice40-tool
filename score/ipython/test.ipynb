{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
