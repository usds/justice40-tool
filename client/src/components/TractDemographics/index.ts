import TractDemographics from './TractDemographics';
export default TractDemographics;
