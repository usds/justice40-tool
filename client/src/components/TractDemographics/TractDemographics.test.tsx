import React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import TractDemographics from './TractDemographics';

const propertiesAS = {
  'GEOID10': '60010950300',
  'SF': 'American Samoa',
  'M_WTR': false,
  'M_WKFC': true,
  'M_CLT': false,
  'M_ENY': false,
  'M_TRN': false,
  'M_HSG': false,
  'M_PLN': false,
  'M_HLTH': false,
  'SM_C': true,
  'SM_DON': false,
  'SM_NO_DON': true,
  'EPLRLI': false,
  'EALRLI': false,
  'EBLRLI': false,
  'PM25LI': false,
  'EBLI': false,
  'DPMLI': false,
  'TPLI': false,
  'LPMHVLI': false,
  'HBLI': false,
  'RMPLI': false,
  'SFLI': false,
  'HWLI': false,
  'WDLI': false,
  'USTLI': false,
  'DLI': false,
  'ALI': false,
  'HDLI': false,
  'LLELI': false,
  'LILHSE': false,
  'PLHSE': false,
  'LMILHSE': false,
  'ULHSE': false,
  'EPL_ET': false,
  'EAL_ET': false,
  'EBL_ET': false,
  'EB_ET': false,
  'PM25_ET': false,
  'DS_ET': false,
  'TP_ET': false,
  'LPP_ET': false,
  'KP_ET': false,
  'HB_ET': false,
  'RMP_ET': false,
  'NPL_ET': false,
  'TSDF_ET': false,
  'WD_ET': false,
  'UST_ET': false,
  'DB_ET': false,
  'A_ET': false,
  'HD_ET': false,
  'LLE_ET': false,
  'UN_ET': false,
  'LISO_ET': false,
  'POV_ET': false,
  'LMI_ET': false,
  'IA_LMI_ET': false,
  'IA_UN_ET': false,
  'IA_POV_ET': true,
  'TC': 1,
  'CC': 1,
  'IAULHSE': false,
  'IAPLHSE': true,
  'IALMILHSE': false,
  'IALMILHSE_PFS': 0.27,
  'IAPLHSE_PFS': 0.98,
  'IAULHSE_PFS': 0.75,
  'LHE': false,
  'IALHE': true,
  'IAHSEF': 0.52,
  'CA_LT20': false,
  'M_CLT_EOMI': false,
  'M_ENY_EOMI': false,
  'M_TRN_EOMI': false,
  'M_HSG_EOMI': false,
  'M_PLN_EOMI': false,
  'M_WTR_EOMI': false,
  'M_HLTH_EOMI': false,
  'M_WKFC_EOMI': true,
  'FPL200S': false,
  'M_WKFC_EBSI': true,
  'TD_ET': false,
  'FLD_ET': false,
  'WFR_ET': false,
  'ADJ_ET': true,
  'ADJ_PFS': 1,
  'IS_ET': false,
  'FUDS_ET': '0',
  'UI_EXP': 'Island Areas',
  'THRHLD': 3,
};

const propertiesNYC = {
  'GEOID10': '36081020800',
  'SF': 'New York',
  'CF': 'Queens County',
  'DF_PFS': 0.69,
  'AF_PFS': 0.46,
  'HDF_PFS': 0.29,
  'DSF_PFS': 0.96,
  'EBF_PFS': 0.75,
  'EBLR_PFS': 0.11,
  'EPLR_PFS': 0.19,
  'HBF_PFS': 0.75,
  'LLEF_PFS': 0.2,
  'LIF_PFS': 0.9,
  'LMI_PFS': 0.35,
  'MHVF_PFS': 0.93,
  'PM25F_PFS': 0.44,
  'HSEF': 0.2830459770114942,
  'P100_PFS': 0.43,
  'P200_PFS': 0.41,
  'P200_I_PFS': 0.37,
  'LPF_PFS': 0.93,
  'KP_PFS': 0.2159833426939357,
  'NPL_PFS': 0.73,
  'RMP_PFS': 0.42,
  'TSDF_PFS': 0.87,
  'TPF': 3136,
  'TF_PFS': 0.92,
  'UF_PFS': 0.86,
  'UST_PFS': 0.9877704355346548,
  'M_WTR': false,
  'M_WKFC': true,
  'M_CLT': false,
  'M_ENY': false,
  'M_TRN': false,
  'M_HSG': false,
  'M_PLN': false,
  'M_HLTH': false,
  'SM_C': true,
  'SM_DON': false,
  'SM_NO_DON': true,
  'EPLRLI': false,
  'EALRLI': false,
  'EBLRLI': false,
  'PM25LI': false,
  'EBLI': false,
  'DPMLI': false,
  'TPLI': false,
  'LPMHVLI': false,
  'HBLI': false,
  'RMPLI': false,
  'SFLI': false,
  'HWLI': false,
  'WDLI': false,
  'USTLI': false,
  'DLI': false,
  'ALI': false,
  'HDLI': false,
  'LLELI': false,
  'LILHSE': true,
  'PLHSE': false,
  'LMILHSE': false,
  'ULHSE': false,
  'EPL_ET': false,
  'EAL_ET': false,
  'EBL_ET': false,
  'EB_ET': false,
  'PM25_ET': false,
  'DS_ET': true,
  'TP_ET': true,
  'LPP_ET': false,
  'HRS_ET': '1',
  'KP_ET': false,
  'HB_ET': false,
  'RMP_ET': false,
  'NPL_ET': false,
  'TSDF_ET': false,
  'WD_ET': false,
  'UST_ET': true,
  'DB_ET': false,
  'A_ET': false,
  'HD_ET': false,
  'LLE_ET': false,
  'UN_ET': false,
  'LISO_ET': true,
  'POV_ET': false,
  'LMI_ET': false,
  'IA_LMI_ET': false,
  'IA_UN_ET': false,
  'IA_POV_ET': false,
  'TC': 1,
  'CC': 1,
  'IAULHSE': false,
  'IAPLHSE': false,
  'IALMILHSE': false,
  'LHE': true,
  'IALHE': false,
  'CA': 0.1,
  'NCA': 0.89,
  'CA_LT20': true,
  'M_CLT_EOMI': false,
  'M_ENY_EOMI': false,
  'M_TRN_EOMI': true,
  'M_HSG_EOMI': false,
  'M_PLN_EOMI': false,
  'M_WTR_EOMI': false,
  'M_HLTH_EOMI': false,
  'M_WKFC_EOMI': true,
  'FPL200S': false,
  'M_WKFC_EBSI': true,
  'TD_ET': false,
  'TD_PFS': 0.61,
  'FLD_PFS': 0.52,
  'WFR_PFS': 0.33,
  'FLD_ET': false,
  'WFR_ET': false,
  'ADJ_ET': true,
  'ADJ_PFS': 1,
  'IS_PFS': 0.99,
  'IS_ET': false,
  'IMP_FLG': '0',
  'DM_B': 0.22,
  'DM_AI': 0.03,
  'DM_A': 0.09,
  'DM_HI': 0,
  'DM_T': 0.1,
  'DM_W': 0.11,
  'DM_H': 0.33,
  'DM_O': 0.36,
  'AGE_10': 0.13,
  'AGE_MIDDLE': 0.8,
  'AGE_OLD': 0.06,
  'UI_EXP': 'Nation',
  'THRHLD': 21,
};

const propertiesAK = {
  'GEOID10': '02290000300',
  'SF': 'Alaska',
  'CF': 'Yukon-Koyukuk Census Area',
  'DF_PFS': 0.87,
  'AF_PFS': 0.93,
  'HDF_PFS': 0.92,
  'DSF_PFS': 0,
  'EBF_PFS': 0.97,
  'EBLR_PFS': 0.76,
  'EPLR_PFS': 0.71,
  'HBF_PFS': 0.22,
  'LLEF_PFS': 0.79,
  'LIF_PFS': 0.5,
  'LMI_PFS': 0.89,
  'MHVF_PFS': 0.07,
  'HSEF': 0.128,
  'P100_PFS': 0.84,
  'P200_PFS': 0.79,
  'P200_I_PFS': 0.81,
  'LPF_PFS': 0.11,
  'KP_PFS': 0.9996164436103616,
  'NPL_PFS': 0,
  'RMP_PFS': 0,
  'TSDF_PFS': 0,
  'TPF': 1821,
  'UF_PFS': 0.97,
  'UST_PFS': 0.0563641031877947,
  'M_WTR': false,
  'M_WKFC': true,
  'M_CLT': false,
  'M_ENY': true,
  'M_TRN': false,
  'M_HSG': false,
  'M_PLN': false,
  'M_HLTH': true,
  'SM_C': true,
  'SM_DON': false,
  'SM_NO_DON': true,
  'EPLRLI': false,
  'EALRLI': false,
  'EBLRLI': false,
  'PM25LI': false,
  'EBLI': true,
  'DPMLI': false,
  'TPLI': false,
  'LPMHVLI': false,
  'HBLI': false,
  'RMPLI': false,
  'SFLI': false,
  'HWLI': false,
  'WDLI': false,
  'USTLI': false,
  'DLI': false,
  'ALI': true,
  'HDLI': true,
  'LLELI': false,
  'LILHSE': false,
  'PLHSE': false,
  'LMILHSE': false,
  'ULHSE': true,
  'EPL_ET': false,
  'EAL_ET': false,
  'EBL_ET': false,
  'EB_ET': true,
  'PM25_ET': false,
  'DS_ET': false,
  'TP_ET': false,
  'LPP_ET': false,
  'KP_ET': true,
  'HB_ET': false,
  'RMP_ET': false,
  'NPL_ET': false,
  'TSDF_ET': false,
  'WD_ET': false,
  'UST_ET': false,
  'DB_ET': false,
  'A_ET': true,
  'HD_ET': true,
  'LLE_ET': false,
  'UN_ET': true,
  'LISO_ET': false,
  'POV_ET': false,
  'LMI_ET': false,
  'IA_LMI_ET': false,
  'IA_UN_ET': false,
  'IA_POV_ET': false,
  'TC': 6,
  'CC': 5,
  'IAULHSE': false,
  'IAPLHSE': false,
  'IALMILHSE': false,
  'LHE': true,
  'IALHE': false,
  'CA': 0.05,
  'NCA': 0.94,
  'CA_LT20': true,
  'M_CLT_EOMI': false,
  'M_ENY_EOMI': true,
  'M_TRN_EOMI': false,
  'M_HSG_EOMI': true,
  'M_PLN_EOMI': true,
  'M_WTR_EOMI': false,
  'M_HLTH_EOMI': true,
  'M_WKFC_EOMI': true,
  'FPL200S': true,
  'M_WKFC_EBSI': true,
  'TD_ET': false,
  'TD_PFS': 0.43,
  'FLD_PFS': 0.05,
  'FLD_ET': false,
  'WFR_ET': false,
  'ADJ_ET': false,
  'ADJ_PFS': 0.75,
  'IS_ET': false,
  'FUDS_ET': '1',
  'IMP_FLG': '0',
  'DM_B': 0,
  'DM_AI': 0.78,
  'DM_A': 0,
  'DM_HI': 0,
  'DM_T': 0.03,
  'DM_W': 0.15,
  'DM_H': 0.02,
  'DM_O': 0,
  'AGE_10': 0.17,
  'AGE_MIDDLE': 0.69,
  'AGE_OLD': 0.12,
  'UI_EXP': 'Nation',
  'THRHLD': 21,
};


describe('rendering of TractDemographics component', () => {
  it('does it render a tract in American Samoa correctly? ', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <TractDemographics properties={propertiesAS}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('does it render a tract in NYC correctly? ', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <TractDemographics properties={propertiesNYC}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('does it render a tract in Alaska correctly? ', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <TractDemographics properties={propertiesAK}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
