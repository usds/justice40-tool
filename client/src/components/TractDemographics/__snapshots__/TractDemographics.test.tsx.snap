// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of TractDemographics component does it render a tract in Alaska correctly?  1`] = `
<DocumentFragment>
  <div>
    <div>
      Tract demographics
    </div>
    <h6>
      Race / Ethnicity
      <span>
        (
        <a
          aria-controls="race-panel"
          aria-expanded="false"
          class="usa-link"
          id="race-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="race-header"
      hidden=""
      id="race-panel"
    >
      <div>
        <span>
          White
        </span>
        <span>
          15%
        </span>
      </div>
      <div>
        <span>
          Black or African American
        </span>
        <span>
          0%
        </span>
      </div>
      <div>
        <span>
          American Indian and Alaska Native
        </span>
        <span>
          78%
        </span>
      </div>
      <div>
        <span>
          Asian
        </span>
        <span>
          0%
        </span>
      </div>
      <div>
        <span>
          Native Hawaiian or Pacific Islander
        </span>
        <span>
          0%
        </span>
      </div>
      <div>
        <span>
          Other
        </span>
        <span>
          0%
        </span>
      </div>
      <div>
        <span>
          Two or more races
        </span>
        <span>
          3%
        </span>
      </div>
      <div>
        <span>
          Hispanic or Latino
        </span>
        <span>
          2%
        </span>
      </div>
    </section>
    <h6>
      Age
      <span>
        (
        <a
          aria-controls="age-panel"
          aria-expanded="false"
          class="usa-link"
          id="age-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="age-header"
      hidden=""
      id="age-panel"
    >
      <div>
        <span>
          Children under 10
        </span>
        <span>
          17%
        </span>
      </div>
      <div>
        <span>
          Ages 10 - 64
        </span>
        <span>
          69%
        </span>
      </div>
      <div>
        <span>
          Elderly over 65
        </span>
        <span>
          12%
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`rendering of TractDemographics component does it render a tract in American Samoa correctly?  1`] = `
<DocumentFragment>
  <div>
    <div>
      Tract demographics
    </div>
    <h6>
      Race / Ethnicity
      <span>
        (
        <a
          aria-controls="race-panel"
          aria-expanded="false"
          class="usa-link"
          id="race-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="race-header"
      hidden=""
      id="race-panel"
    >
      <div>
        <span>
          White
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Black or African American
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          American Indian and Alaska Native
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Asian
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Native Hawaiian or Pacific Islander
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Other
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Two or more races
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Hispanic or Latino
        </span>
        <span>
          --
        </span>
      </div>
    </section>
    <h6>
      Age
      <span>
        (
        <a
          aria-controls="age-panel"
          aria-expanded="false"
          class="usa-link"
          id="age-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="age-header"
      hidden=""
      id="age-panel"
    >
      <div>
        <span>
          Children under 10
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Ages 10 - 64
        </span>
        <span>
          --
        </span>
      </div>
      <div>
        <span>
          Elderly over 65
        </span>
        <span>
          --
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`rendering of TractDemographics component does it render a tract in NYC correctly?  1`] = `
<DocumentFragment>
  <div>
    <div>
      Tract demographics
    </div>
    <h6>
      Race / Ethnicity
      <span>
        (
        <a
          aria-controls="race-panel"
          aria-expanded="false"
          class="usa-link"
          id="race-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="race-header"
      hidden=""
      id="race-panel"
    >
      <div>
        <span>
          White
        </span>
        <span>
          11%
        </span>
      </div>
      <div>
        <span>
          Black or African American
        </span>
        <span>
          22%
        </span>
      </div>
      <div>
        <span>
          American Indian and Alaska Native
        </span>
        <span>
          3%
        </span>
      </div>
      <div>
        <span>
          Asian
        </span>
        <span>
          9%
        </span>
      </div>
      <div>
        <span>
          Native Hawaiian or Pacific Islander
        </span>
        <span>
          0%
        </span>
      </div>
      <div>
        <span>
          Other
        </span>
        <span>
          36%
        </span>
      </div>
      <div>
        <span>
          Two or more races
        </span>
        <span>
          10%
        </span>
      </div>
      <div>
        <span>
          Hispanic or Latino
        </span>
        <span>
          33%
        </span>
      </div>
    </section>
    <h6>
      Age
      <span>
        (
        <a
          aria-controls="age-panel"
          aria-expanded="false"
          class="usa-link"
          id="age-header"
          tabindex="0"
        >
          show
        </a>
        <img
          alt="expand icon"
          src="test-file-stub"
        />
        )
      </span>
    </h6>
    <section
      aria-labelledby="age-header"
      hidden=""
      id="age-panel"
    >
      <div>
        <span>
          Children under 10
        </span>
        <span>
          13%
        </span>
      </div>
      <div>
        <span>
          Ages 10 - 64
        </span>
        <span>
          80%
        </span>
      </div>
      <div>
        <span>
          Elderly over 65
        </span>
        <span>
          6%
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
